packages/fuels-accounts/src/provider.rs

```diff
@@ -256,6 +257,31 @@ impl Provider {
         Ok(self.client.estimate_gas_price(block_horizon).await?)
     }
 
+    async fn estimate_predicates_with_node_fallback<T: Transaction>(
+        &self,
+        tx: &mut T,
+    ) -> Result<()> {
+        if self.latest_chain_executor_version().await? > LATEST_STATE_TRANSITION_VERSION {
```

Comment:
This calls the node for `chain_info` to get the executor version every time a transaction is estimated. A much better approach (suggested by @segfault-magnet) would be to cache the on-chain executor version (or the whole `chain_info`) and only update it periodically.

Reply:
Taking `chain_info` from `prepare_transaction_for_sending` - no extra network calls, no need for caching (yet).